lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/time/Time.java

```diff
@@ -18,53 +18,63 @@
  */
 package org.apache.reef.wake.time;
 
+import java.util.Date;
+
 /**
- * Time object.
+ * An abstract object that has a timestamp.
+ * That allows us to compare and order objects by the time.
  */
 public abstract class Time implements Comparable<Time> {
 
   private final long timestamp;
 
+  /**
+   * Initialize the internal timestamp. Timestamp remains constant
+   * for the entire lifecycle of the object.
+   * @param timestamp timestamp in milliseconds since the beginning
+   * of the epoch (01/01/1970).
+   */
   public Time(final long timestamp) {
     this.timestamp = timestamp;
   }
 
+  /**
+   * Get timestamp in milliseconds since the beginning of the epoch (01/01/1970).
+   * @return Object's timestamp in milliseconds since the start of the epoch.
+   */
+  public final long getTimestamp() {
+    return this.timestamp;
+  }
+
+  /**
+   * Get timestamp in milliseconds since the beginning of the epoch (01/01/1970).
+   * @return Object's timestamp in milliseconds since the start of the epoch.
+   * @deprecated [REEF-1532] Prefer using getTimestamp() instead.
+   * Remove after release 0.16.
+   */
   public final long getTimeStamp() {
     return this.timestamp;
   }
 
   @Override
-  public final String toString() {
-    return this.getClass().getName() + "[" + this.timestamp + "]";
+  public String toString() {
+    return this.getClass().getName()
+        + ":[" + this.timestamp + '|' + new Date(this.timestamp) + ']';
   }
 
   @Override
-  public final int compareTo(final Time o) {
-    if (this.timestamp < o.timestamp) {
-      return -1;
-    }
-    if (this.timestamp > o.timestamp) {
-      return 1;
-    }
-    if (this.hashCode() < o.hashCode()) {
-      return -1;
-    }
-    if (this.hashCode() > o.hashCode()) {
-      return 1;
-    }
-    return 0;
+  public int compareTo(final Time other) {
+    final int cmp = Long.compare(this.timestamp, other.timestamp);
+    return cmp != 0 ? cmp : Integer.compare(this.hashCode(), other.hashCode());
```

Comment:
From what I see this.hashCode() will use the default hashCode() implementation, right? So two Time objects will be equal only if they are the same instance. Is this the desired behavior? if yes then wouldn't it be better to implement equals just by comparing references?